test/unit/strict-template-policy.html

```diff
@@ -0,0 +1,240 @@
+      // wraps Chai's assert.throws to re-throw uncaught errors
+      function assertThrows(fn, re) {
+        // Safari does not forward error messages
+        re = new RegExp('(' + re.toString().slice(1,-1) + ')|(Script error\\.)', re.flags);
+        // Catch uncaught errors; note when running in iframe sometimes
+        // Safari errors are thrown on the top window, sometimes not, so
+        // catch in both places
+        let uncaughtError = null;
+        window.onerror = window.top.onerror = function(err) {
+          if (!uncaughtError) {
+            uncaughtError = new Error(err);
+          }
+        };
+        assert.throws(function() {
+          fn();
+          // Re-throw any uncaughtErrors
+          let err;
+          if ((err = (uncaughtError || window.uncaughtError))) {
+            throw new Error(err.message);
+          }
+          // Force polyfill reactions and/or async template stamping
+          Polymer.flush();
+          // Re-throw any uncaughtErrors
+          if ((err = (uncaughtError || window.uncaughtError))) {
+            throw new Error(err.message);
+          }
+        }, re);
+      }
+
+      test('dom-bind', function() {
+        assertThrows(function() {
+          document.getElementById('target').innerHTML =
+          '<dom-bind>' +
+          '  <template>' +
+          '    <div id="injected"></div>'+
+          '  </template>`' + 
+          '</dom-bind>';
+        }, /dom-bind not allowed/);
+        assert.notOk(document.getElementById('injected'));
+      });
+
+      test('dom-if', function() {
+        assertThrows(function() {
+          document.getElementById('target').innerHTML =
+            '<dom-if if>' +
+            '  <template>' +
+            '    <div id="injected"></div>'+
+            '  </template>' + 
+            '</dom-if>';
+        }, /template owner not trusted/);
+        assert.notOk(document.getElementById('injected'));
+      });
+
+      test('dom-repeat', function() {
+        assertThrows(function() {
+          document.getElementById('target').innerHTML =
+            '<dom-repeat items="[0]">' +
+            '  <template>' +
+            '    <div id="injected"></div>'+
+            '  </template>`' + 
+            '</dom-repeat>';
+        }, /template owner not trusted/);
+        assert.notOk(document.getElementById('injected'));
+      });
+
+      test('dom-module after registration', function() {
+        assertThrows(function() {
+          document.getElementById('target').innerHTML =
+            '<dom-module id="trusted-element">' +
+            '  <template>' +
+            '    <div id="injected"></div>'+
+            '  </template>`' +
+            '</dom-module>' +
+            '<trusted-element></trusted-element>';
+        }, /trusted-element registered twice/);
+        const el = document.querySelector('trusted-element');
+        assert.notOk(el && el.shadowRoot && el.shadowRoot.querySelector('#injected'));
```

Comment:
Remove `&& el.shadowRoot.querySelector('#injected')` as it's enough to assert that the element does not have a shadowRoot when the template gets nulled due to 2x register.

Reply:
Separate the creation of the `trusted-element` so an innerHTML error does not disrupt element creation.

Reply:
Done & done.